### PR TITLE
Add generic SERDE support

### PR DESCRIPTION
**Pull request by rnpridgeon**
Precursor to Schema Registry support

### CODE REVIEW

File: examples/protobuf_example/consumer/protobuf_example.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+			run = false
+		default:
+			ev := c.Poll(100)
+			if ev == nil {
+				continue
+			}
+
+			switch e := ev.(type) {
+			case *kafka.Message:
+				fmt.Printf("%% Message on %s:\n%s\n",
```
Comment (edenhill) [inline examples/protobuf_example/consumer/protobuf_example.go]: it should grab and use the deserialized object here

File: examples/protobuf_example/producer/protobuf_example.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	"github.com/golang/protobuf/proto"
+	"log"
+	"os"
+)
+
+<<<<<<< Updated upstream
+var values  = []proto.Message{
+=======
+var values = []proto.Message{
+>>>>>>> Stashed changes
```
Comment (cricket007) [inline examples/protobuf_example/producer/protobuf_example.go]: Git diffs in here

File: kafka/config.go [hunk 3]
```diff
@@ -213,6 +214,28 @@ func (m ConfigMap) get(key string, defval ConfigValue) (ConfigValue, error) {
...
+	log.Printf("Extracting %s\n", m2)
+	return m2
+}
+
+// Extract performs a get() and if found deletes the key.
+func (m ConfigMap) Extract(key string, defval ConfigValue) (ConfigValue, error) {
+	return m.extract(key, defval)
+}
+
+>>>>>>> Stashed changes
```
Comment (cricket007) [inline kafka/config.go]: Git diffs

File: kafka/error.go [hunk 4]
```diff
@@ -24,13 +24,28 @@ package kafka
...
+=======
+// NewError creates a new pointer to an Error instance with an optional message
+func NewError(code ErrorCode, str error) *Error {
+	return &Error{
+		code: code,
+		str: fmt.Sprintf("%s: %s", code, str),
+	}
+}
+
+>>>>>>> Stashed changes
```
Comment (cricket007) [inline kafka/error.go]: Git diffs

**Comment by rnpridgeon** [comment]
Thanks @cricket007 , I'm actually closing this one out. I was using it to quickly share the interface with a few others. It will be part of an overall SR merge. I'm about to change some changes including complete examples to the `avro` branch. Please do share your thoughts there as well once I push the changes.

**Pull request closed without merging**